ggml/src/ggml-cuda/cpy-utils.cuh

```diff
@@ -2,26 +2,20 @@
 
 #include "ggml-common.h"
 
-static __device__ __forceinline__ void convert_f32_f32(const float * src, float * dst) {
-    *dst = *src;
+template<typename src_t, typename dst_t>
+static __device__ __forceinline__ void convert_to_flt(const src_t * src, dst_t * dst) {
+    *dst = float(*src);
 }
 
-static __device__ __forceinline__ void convert_f32_f16(const float * src, half * dst) {
+template<typename src_t>
+static __device__ __forceinline__ void convert_to_f16(const src_t * src, half * dst) {
     *dst = __float2half(*src);
```

Comment:
Just to make sure: did you try the generic version for this case too and it failed to compile?